src/analysis/calculus/inverse.lean

```diff
@@ -363,6 +379,32 @@ protected lemma inj_on (hf : approximates_linear_on f (f' : E →L[𝕜] F) s c)
   inj_on f s :=
 inj_on_iff_injective.2 $ hf.injective hc
 
+protected lemma surjective [complete_space E]
+  (hf : approximates_linear_on f (f' : E →L[𝕜] F) univ c) (hc : subsingleton E ∨ c < N⁻¹) :
+  surjective f :=
+begin
+  cases hc with hE hc,
+  { haveI : subsingleton F := (equiv.subsingleton_congr f'.to_linear_equiv.to_equiv).1 hE,
+    exact surjective_to_subsingleton _ },
+  { suffices H : ∀ (R : ℝ), 0 ≤ R → closed_ball (f 0) R ⊆ range f,
+    { have : (⋃ (n : ℕ), closed_ball (f 0) n) ⊆ range f :=
+        Union_subset (λ n, H n (nat.cast_nonneg _)),
+      rw Union_closed_ball_nat at this,
```

Comment:
It's a little awkward introducing this unnecessary reduction to `ℕ`.  Would you mind writing an "induction principle for metric spaces" of the form
```lean
example {α : Type*} (p : α → Prop) [metric_space α] {x : α}
  (H : ∃ᶠ (R : ℝ) in at_top, ∀ y ∈ closed_ball x R, p y) (y : α) :
  p y :=
```